compare_platforms/compare_platforms_read_only.c

```diff
@@ -0,0 +1,147 @@
+/** 
+ * This version can be run in both a CHERI and a non-CHERI environment to 
+ * compare the two.
+ * 
+ * In a CHERI environment, it will:
+ *     Step 1: Use can_read to check if the struct's "permissions" member includes write permissions.
+ *     Step 2: Use capabilities to enforce read-only permissions
+ *             (see https://github.com/capablevms/cheri-examples/blob/master/employee/include/employee.h).
+ *     Step 3: Try to write to the read-only struct members, and SIGPROT.
+ *
+ * In a non-CHERI environment, it will:
+ *     Step 1: Use can_read to check if the struct's "permissions" member includes write permissions.
+ *     Step 2: Try to exceed the bounds of an object and add write permissions
+ *             (see https://ctsrd-cheri.github.io/cheri-exercises/exercises/buffer-overflow-heap/index.html).
+ *     Step 3: Rewrite the review.
+ *
+ * In a CHERI environment, the overflow attack will be unreachable.
+ * In morello-purecap, it would crash immediately whether or not the struct
+ * had been set as read only.
+ *
+ * In a non-CHERI environment, this attack should succeed in corrupting some of the data.
+ *
+ */
+
+#include "include/review.h"
+#include <assert.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+int main()
+{
+    char *username, *realname, *privatereview, *publicreview, *permissions;
+    
+    // Our new review, which will have read only permissions.
+    struct review review;
+    
+    size_t smallsz = 0x20;
+    size_t biggersz = 0x1001;
```

Comment:
What do these two values represent?

Reply:
Malloc amounts for the shorter struct members (names) and longer ones (review text).

Reply:
Could we make them `const` and add a small comment like "initialization values for struct member sizes"?

Reply:
Fixed in cdc071e3ab3a1248c190f23097a4e8a14dd40167